Tests/RevenueCatUITests/CustomerCenter/ManageSubscriptionsViewModelTests.swift

```diff
@@ -71,27 +73,256 @@ class ManageSubscriptionsViewModelTests: TestCase {
+
+        // Act
+        await viewModel.loadScreen()
+
+        // Assert
+        expect(viewModel.subscriptionInformation).toNot(beNil())
+        expect(viewModel.screen).toNot(beNil())
+        expect(viewModel.state) == .success
+
+        expect(viewModel.subscriptionInformation?.title) == "yearly"
+        expect(viewModel.subscriptionInformation?.durationTitle) == "year"
+        expect(viewModel.subscriptionInformation?.price) == "$29.99"
+
+        let formatter = DateFormatter()
+        formatter.dateStyle = .medium
+        expect(viewModel.subscriptionInformation?.nextRenewalString) ==
+            formatter.string(from: ISO8601DateFormatter().date(from: expirationDateFirst)!)
+        expect(viewModel.subscriptionInformation?.productIdentifier) == productIdOne
+    }
+
+    func testShouldShowAppleSubscription_whenUserHasBothGoogleAndAppleSubscriptions() async {
+        // Arrange
+        let productIdOne = "com.revenuecat.product1"
+        let productIdTwo = "com.revenuecat.product2"
+        let purchaseDate = "2022-04-12T00:03:28Z"
+        let expirationDateFirst = "2062-04-12T00:03:35Z"
+        let expirationDateSecond = "2062-05-12T00:03:35Z"
+        let products = [
+            Fixtures.product(id: productIdOne, title: "yearly", duration: .year, price: 29.99),
+            Fixtures.product(id: productIdTwo, title: "monthly", duration: .month, price: 2.99)
+        ]
+        let customerInfo = Fixtures.customerInfo(
+            subscriptions: [
+                Fixtures.Subscription(
+                    id: productIdOne,
+                    store: "play_store",
+                    purchaseDate: purchaseDate,
+                    expirationDate: expirationDateFirst
+                ),
+                Fixtures.Subscription(
+                    id: productIdTwo,
+                    store: "app_store",
+                    purchaseDate: purchaseDate,
+                    expirationDate: expirationDateSecond
+                )
+            ].shuffled(),
+            entitlements: [
+                Fixtures.Entitlement(
+                    entitlementId: "premium",
+                    productId: productIdOne,
+                    purchaseDate: purchaseDate,
+                    expirationDate: expirationDateFirst
+                ),
+                Fixtures.Entitlement(
+                    entitlementId: "plus",
+                    productId: productIdTwo,
+                    purchaseDate: purchaseDate,
+                    expirationDate: expirationDateSecond
+                )
+            ].shuffled()
+        )
+
+        let viewModel = ManageSubscriptionsViewModel(screen: ManageSubscriptionsViewModelTests.screen,
+                                                     purchasesProvider: MockManageSubscriptionsPurchases(
+                                                        customerInfo: customerInfo,
+                                                        products: products
```

Comment:
haven't looked in detail that the class, but this is the first thing that really jumps out at me as not very swiftUI-ish, the viewModel and the fact that it holds a reference to the screen. 

Why do we hold that reference? Typically you'd have the viewModel own the properties, and have those be published, then a View (or many) observes them

Reply:
Totally agree. In MVVM the ViewModel should be completely decoupled from the UI layer and just publish state, not caring about who listens. 

I did not change this ViewModel, but I can take a look how it can be improved, maybe in a follow-up PR. 

I see that in this case, the `Screen` class is not a SwiftUI view, but a `struct` holding some input data. Also, it is unused. @vegaro Are there any future plans for this, or can it be removed?

Reply:
Oh this is confusing, but screen is the data of the screen that is being sent down in the JSON, so the screen properties like screen title, feedback survey options, etc, it doesn't have anything to do with view classes. It's been used in upward branches so it can't be removed, for example https://github.com/RevenueCat/purchases-ios/pull/3968/files#diff-121dcc7927012032acdbbf026716bf9daf66d91cee530ab12eae052d06526aa0R187

Reply:
Maybe there's a better name for the struct, to avoid the confusion. Something like `Destination` maybe? 

Also I don't see it being used still, but maybe I'm looking in the wrong place haha. It looks like it's only being set, never read, from what I can tell.